src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    private side1: number,
+    private side2: number,
+    private side3: number,
+  ) {
+    if (Math.min(side1, side2, side3) <= 0) {
+      throw new Error('Sides must be greater than zero');
+    }
+
+    if (this.isNotTriangle()) {
+      throw new Error('Sum of cathetus must be greater than hypotenuse');
+    }
+  }
+
+  isNotTriangle(): boolean {
+    const sides = [this.side1, this.side2, this.side3]
+      .sort((a, b) => a - b);
+    const [cat1, cat2, hip] = sides;
 
-export class Circle {
+    return cat1 + cat2 <= hip;
+  }
 
+  getArea(): number {
+    const sides = [this.side1, this.side2, this.side3].sort();
```

Comment:
You don't have to sort this. Semiperimeter will be greater than any of the sides.

For any a, b, c that forms a triangle

a + b > c,
a + b + c > 2c
(a + b + c) / 2 > c
semiperimeter > c
semiperimeter > b
semiperimeter > a

cus same true for 
a + c > b,
b + c > a,

Reply:
✔️